react-ui/src/pages/contest/Contest.jsx

```diff
@@ -306,6 +137,7 @@ function Contest() {
     }
```

Comment:
From what I can tell, this `useEffect` never fires with the modal, so the page doesn't scroll down to the last viewed entry. Compare opening an entry, scrolling a dozen entries or so away and then clicking the back button on the page. On prod, the contest page will scroll to the last viewed entry, but that isn't working in this code.

It's also not resetting the state once it's used, so if you view an entry modal, and then navigate back to the contest page, it doesn't scroll or reset scroll state. So when you navigate out of the modal and then out of the contest page, and then revisit the contest page, it scrolls to the last entry.

Reply:
> but that isn't working in this code.
Can you clarify what you are seeing? When I run this code, opening and closing the modal does not change the `ContestGrid` scroll position at all.

> it scrolls to the last entry
Oddly, the behavior is different on Firefox vs Chrome and Safari. I'll dig into this a bit more.

Reply:
Removed scroll logic for a few reasons:
1. Scroll logic is no longer required with modal. The modal should appear above the contest page without disturbing
2. Scroll logic is not really required for direct link. Direct link is pretty rare, and It's totally reasonable to start at the top when exiting. Plus, none of the underlying flags are in position yet.
The PR I'm about to push *should* fix both issues, but we do need to go through that useEffect still and determine if any other code is unnecessary or at least give some comments.

Reply:
1. The modal is still a full page modal. While the component has changed, the end user experience hasn't changed. Being able to navigate several entries away and then return back to the contest page and be back at the top of the page doesn't feel like the best experience IMO. Let's say it's after the results, you click on the winner and then navigate to see the first 10-20 entries and decide now that the quality is lower, you just want to see the rest in the gallery. You can currently click back and be right at the last one you viewed, but without scrolling update you have to scroll back down. Or if you accidentally click back, you have to find the last flag you were looking for. Minor scenarios, but what's the benefit of no longer supporting these use cases?

2. Yeah, it's probably not absolutely necessary, but the underlying flags may be in position since the contest results are cached. So if you open an entry in a new tab and then click back in that tab, the gallery is in the same state and the user has seen it before. Not saying that this is necessarily something that we have, or even should, address, just pointing out the scenario. I also wouldn't necessarily say that direct link is rare, I've certainly shared direct entry links back and forth with a friend when we're discussing our favorite entries, so I imagine some other folks may do the same. We've also shared some in the mod channel when discussing certain entries as well. I do agree that there doesn't necessarily need to be an expectation to scroll once you exit out of a direct link since that is a new session.

TL;DR - What's the goal of removing the scroll logic? I feel like it provides a somewhat better user experience. I agree that updating scroll position on direct links isn't necessary.

Reply:
Ah ok, sorry I hadn't considered the case where the user arrows between the entries and then exits. Unfortunately, I cannot follow some of the useEffect logic, so we'd need to set aside some time to step through it over Discord or you can take a crack at it. Let me know which you prefer.

Reply:
I think let's first walk through it together, that way you can see what makes sense and possibly suggest improvements.